src/Bicep.Core/Parsing/Parser.cs

```diff
@@ -1015,6 +1015,14 @@ private SyntaxBase ObjectProperty()
                     return this.NewLine();
                 }
 
+                // Nested resource declarations may be allowed
+                if (allowResourceDeclarations && 
+                        (current.Type == TokenType.At || 
+                        (current.Type == TokenType.Identifier && current.Text == LanguageConstants.ResourceKeyword)))
+                {
+                    return Declaration();

```

Comment:
>Declaration [](start = 27, length = 11)

You should update `ObjectSyntax` to expose resource declarations as a separate properties similar to `Properties`.

Reply:
Would that still be useful given the addition of decorators/conditions/loops? There's 4 different types to check, and it would end up returning SyntaxBase?

(I didn't have a need for this as part of the PR)

Reply:
We want to AST to be easily queryable outside of core compiler scenarios. For example, it would be useful for a linter rule author to just get the list of child resources. In general it just boils down to an extra property that returns `mainImmutableListOfThings.OfType<WhateverSYntax>()`

I still find it interesting that it wasn't needed. I'll look at the latest commits...

Reply:
Talked offline. This makes sense to be now, I was confused about the syntax node structure when loops and ifs were involved.